src/Workspaces/Core/Portable/Storage/SQLite/v1/Interop/SqlConnection.cs

```diff
@@ -61,53 +60,37 @@ public static SqlConnection Create(IPersistentStorageFaultInjector faultInjector
             // one is only used from a single thread at a time.
             // see https://sqlite.org/threadsafe.html for more detail
             var flags = OpenFlags.SQLITE_OPEN_CREATE | OpenFlags.SQLITE_OPEN_READWRITE | OpenFlags.SQLITE_OPEN_NOMUTEX;
-            var result = (Result)raw.sqlite3_open_v2(databasePath, out var handle, (int)flags, vfs: null);
+            var handle = NativeMethods.sqlite3_open_v2(databasePath, (int)flags, vfs: null, out var result);
 
             if (result != Result.OK)
             {
+                handle.Dispose();
```

Comment:
this personally confuses me.  the impl of sqlite3_open_v2 specifically gives back a bogus handle in the case of failure.  so in that case, disposing it just serves to do no work.  To me, it seems preferable to just return a `handle?`.  client then knows they have to check that, and deal with the null case (which certainly wouldn't need to be disposed).  Or, it succeeds, and you have a real handle you cangrab and pass along.

Reply:
> this personally confuses me. the impl of sqlite3_open_v2 specifically gives back a bogus handle in the case of failure.

This precisely follows the behavior that would occur if the P/Invoke signatures were originally written to operate on safe handles instead of `IntPtr` (which is the preferable way to write the signatures since .NET 2.0).

Reply:
that's fine.  i would personally prefer docs here then.  because when i do the computation/data-flow in my head it basically indicates a smell here: i.e. i'm making a call that is *certain* to do nothing.  if that's appropriate, we should at least doc.  That way when i'm reading this 3 years from now i realize this is intentional and isn't just a line of code i'll throw out when doing cleanup :)

Reply:
Added docs

Reply:
i don't see the docs.  can you push?

Reply:
They are in de40b567a507b1a2c7fe5a36d33444e059c9b89b